docs/enhancements/multi-rhel-fips.md

```diff
@@ -0,0 +1,137 @@
+---
+title: fips-with-multiple-rhel-versions
+authors:
+  - "@carbonin"
+creation-date: 2024-05-07
+last-updated: 2024-05-08
+---
+
+# Support FIPS for installers built for different RHEL releases
+
+## Summary
+
+In order for an OpenShift cluster to be considered FIPS compliant the installer
+must be run on a system with FIPS compliant openssl libraries. This means using
+a dynamically linked `openshift-install` binary against the openssl libraries
+present on our container image. Today this is not a problem because all
+`openshift-install` binaries in use have been expecting to link to RHEL 8 based
+openssl libraries, but now OpenShift 4.16 will ship and installer that requires
+RHEL 9 libraries.
```

Comment:
>The second sentence seems to imply that there might be a statically-linked binary available, or that it's an option, but is that the case?

Yes, there is. Actually I'm working on a bug right now that is solved by using the `openshift-install` binary from 4.16. https://issues.redhat.com/browse/OCPBUGS-33227

That said, it is that simple. I just figured it would be easier to do this split in any case.
If we also want to consider the static binary for non-fips cases we just add complexity (3 cases instead of just 2)

Reply:
Where would someone obtain a statically vs dynamically linked build?

Reply:
As of https://issues.redhat.com/browse/CORS-3418 `openshift-install` is statically linked and as of https://issues.redhat.com/browse/CORS-3419 it supports baremetal installs.

`openshift-baremetal-install` is now only intended for use with FIPS use-cases as that's the only time you'd need a dynamically linked binary. https://issues.redhat.com/browse/WRKLDS-1171 added a new command to `oc release extract` so it could be referenced as `openshift-install-fips` which more accurately represents its purpose.